new/detector/implementation/javascript/property/property.go

```diff
@@ -0,0 +1,120 @@
+package property
+
+import (
+	"fmt"
+
+	"github.com/bearer/curio/new/detector/types"
+	"github.com/bearer/curio/new/language/tree"
+	"github.com/rs/zerolog/log"
+
+	generictypes "github.com/bearer/curio/new/detector/implementation/generic/types"
+	languagetypes "github.com/bearer/curio/new/language/types"
+)
+
+type propertyDetector struct {
+	pairQuery         *tree.Query
+	functionNameQuery *tree.Query
+	methodNameQuery   *tree.Query
+}
+
+func New(lang languagetypes.Language) (types.Detector, error) {
+	// { user: "admin_user" }
+	pairQuery, err := lang.CompileQuery(`(pair key: (_) @key value: (_) @value) @root`)
+	if err != nil {
+		return nil, fmt.Errorf("error compiling pair query: %s", err)
+	}
+
+	// function getName(){}
+	functionNameQuery, err := lang.CompileQuery(`(function_declaration name: (identifier) @name) @root`)
+	if err != nil {
+		return nil, fmt.Errorf("error compiling function name query: %s", err)
+	}
+
+	// class User {
+	//	constructor(name, surname)
+	//	GetName()
+	// }
+	methodNameQuery, err := lang.CompileQuery(`(method_definition name: (property_identifier) @name (formal_parameters) @params ) @root`)
+	if err != nil {
+		return nil, fmt.Errorf("error compiling class method query: %s", err)
+	}
+
+	return &propertyDetector{
+		pairQuery:         pairQuery,
+		functionNameQuery: functionNameQuery,
+		methodNameQuery:   methodNameQuery,
+	}, nil
+}
+
+func (detector *propertyDetector) Name() string {
+	return "property"
+}
+
+func (detector *propertyDetector) DetectAt(
+	node *tree.Node,
+	evaluator types.Evaluator,
+) ([]interface{}, error) {
+	// run pair query
+	result, err := detector.pairQuery.MatchOnceAt(node)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result) != 0 {
+		key := result["key"]
+		// { user: "admin_user"} || {"user": "admin_user"}
+		if key.Type() == "property_identifier" || key.Type() == "string" {
+			return []interface{}{generictypes.Property{Name: result["key"].Content()}}, nil
```

Comment:
Do we need to strip quotes for the string case?

Reply:
nice catch, I've added code to handle it as well as test case